### PR TITLE
[DylibBinder] Adding tests

### PR DESCRIPTION
**Pull request by tj-devel709**
This PR runs the DynamicXmlTests using the DylibBinder for reflection!

- Passes 34 Tests
- Fails 57 Tests 

A handful of tests fail for small things like order of items or the public name being incorrect.
I also redid some things in DylibBinder to fix major issues such as global functions being accounted for.
Each test that fails is linked to an issue.

Issue containing large list of things we cannot do in DylibBinder yet: https://github.com/xamarin/binding-tools-for-swift/issues/697
Generic Parameter Issue: https://github.com/xamarin/binding-tools-for-swift/issues/702
Constructor Behaviour Issue: https://github.com/xamarin/binding-tools-for-swift/issues/699
Protocols not fully supported Issue: https://github.com/xamarin/binding-tools-for-swift/issues/698
Destructor Issue: https://github.com/xamarin/binding-tools-for-swift/issues/700
Convenience vs Designated Constructor Issue: https://github.com/xamarin/binding-tools-for-swift/issues/701

### CODE REVIEW

File: DylibBinder/CheckInventory.cs [hunk 1]
```diff
@@ -56,5 +54,19 @@ void GetProtocols (ModuleInventory mi, SwiftName module)
...
 		}
+
+		public static List<OverloadInventory> GetGlobalFunctions (ModuleInventory mi, SwiftName module)
+		{
+			Exceptions.ThrowOnNull (mi, nameof (mi));
+			Exceptions.ThrowOnNull (module, nameof (module));
+			var functions = new List<OverloadInventory> ();
+
+			foreach (var f in mi.FunctionsForName (module)) {
+				if (!f.Name.Name.Contains ("_")) {
```
Comment (stephen-hawley) [inline DylibBinder/CheckInventory.cs:65]: what will happen with:
```
public func this_is_cool() -> Int {
    return 42
}
```
?

File: DylibBinder/DBFunc.cs [hunk 2]
```diff
@@ -32,7 +32,7 @@ internal class DBFunc : IAssociatedTypes {
 		public DBGenericParameters GenericParameters { get; }
 		public DBAssociatedTypes AssociatedTypes { get; } = new DBAssociatedTypes ();
 
-		public DBFunc (TLFunction tlf, bool isMutating = false, bool isProtocol = false)
+		public DBFunc (TLFunction tlf, bool isMutating = false, bool isProtocol = false, bool isGlobal = false)
```
Comment (stephen-hawley) [inline DylibBinder/DBFunc.cs:35]: This function shouldn't need an `isGlobal` flag. You should get the same result from doing `tlf.IsTopLevelFunction`.
The `TLFunction` type is a holder for functions defined in a dylib. Basically, the `TL` stands for top level, but what happens when you get a member function? Then the member is defined on a class and the `TLFunction` will have a class associated with it and with a nesting name length > 0. This predicate checks that for you.

**Review by stephen-hawley — CHANGES_REQUESTED** [review]

File: DylibBinder/CheckInventory.cs (hunk 1 above)
Comment (tj-devel709) [reply DylibBinder/CheckInventory.cs]: I realized that I made an extension method in StringExtensions called `IsPublic (this string s)` that I use elsewhere in this file for the same purpose. I made sure all 3 times, I use this method. Then I linked this issue in that method.

In the issue I included the private and internal classes, enums, structs, protocols, and functions that have the underscore. I need to find another way to fix that!

Comment (stephen-hawley) [reply DylibBinder/CheckInventory.cs]: This should be doable without needing to look at the name. I'll talk to you about this directly.

**Pull request merged**